### PR TITLE
Feature/2229 add invitations to capability create form

### PR DESCRIPTION
**Pull request by HrisyToteva**
closes dfds/cloudplatform/issues/2229

# Additional Review Notes
You invite AD users to on creation of a capability. 

image

The already picked users are stored below the invitees input text: 

image

**Note: This is not perfect it is an initial state there is a lot room for improvements**

### CODE REVIEW

File: src/src/components/DropDownMenu/index.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import style from "./dropdownmenu.module.css";
+import ProfilePicture from "pages/capabilities/members/profilepicture";
+
+export default function DropDownMenu({
```
Comment (kralle333) [inline src/src/components/DropDownMenu/index.js:5]: Maybe change it to a less generic name, sounds like its a DropDownMenu you can use for anything, but in reality it is for invitations

File: src/src/hooks/Capabilities.js [hunk 2]
```diff
@@ -15,19 +15,21 @@ export function useCapabilities() {
...
         name: name,
         description: description,
+        invitees: invitations,
       },
     });
   };
 
   useEffect(() => {
     if (addedCapability) {
+      console.log(addedCapability);
```
Comment (kralle333) [inline src/src/hooks/Capabilities.js:32]: remove this :)

File: src/src/pages/capabilities/NewCapabilityDialog.js [hunk 3]
```diff
@@ -1,8 +1,18 @@
...
+import { getUsers } from "GraphApiClient";
+import DropDownMenu from "components/DropDownMenu";
+
+function TextWrapper() {
+  return (
+    <>
+      <div className=""></div>
+    </>
+  );
+}
```
Comment (kralle333) [inline src/src/pages/capabilities/NewCapabilityDialog.js:15]: is this really needed?

File: src/src/pages/capabilities/NewCapabilityDialog.js [hunk 4]
```diff
@@ -39,6 +69,18 @@ export default function NewCapabilityDialog({
     setFormData((prev) => ({ ...prev, ...{ description: newValue } }));
   };
 
+  async function changeInvitation(e) {
+    e.preventDefault();
+    const regex = /[^,]*$/;
```
Comment (kralle333) [inline src/src/pages/capabilities/NewCapabilityDialog.js:74]: maybe a comment what this regex does?

File: src/src/pages/capabilities/NewCapabilityDialog.js [hunk 5]
```diff
@@ -39,6 +69,18 @@ export default function NewCapabilityDialog({
...
+    setIsUserSearchActive(true);
+    const adValue = e?.target?.value.match(regex)[0];
+    const value = e?.target?.value;
```
Comment (kralle333) [inline src/src/pages/capabilities/NewCapabilityDialog.js:77]: maybe rename these to reflect what the type of the values are. Value is very generic and from reading the code I don't know the type.

File: src/src/pages/capabilities/NewCapabilityDialog.js [hunk 6]
```diff
@@ -63,7 +105,19 @@ export default function NewCapabilityDialog({
...
   const handleAddCapabilityClicked = () => {
     if (onAddCapabilityClicked) {
-      onAddCapabilityClicked(formData);
+      onAddCapabilityClicked({ ...formData, invitations: invitees });
+    }
+  };
+
+  const OnKeyEnter = (e) => {
+    if (e.key === "Enter") {
+      if (!Array.isArray(formData.invitations)) {
```
Comment (kralle333) [inline src/src/pages/capabilities/NewCapabilityDialog.js:114]: reverse statements to reduce nesting

**Review by kralle333 — DISMISSED** [review]
Veerry cool! Good job 😄 

Only got some smaller comments

**Review by SEQUOIIA — APPROVED** [review]
Seems good, also @kralle333 's concerns were addressed in https://github.com/dfds/selfservice-portal/pull/133/commits/cb4b6bfc44c8d7dff3c4ea8e1c66e2dc416c35bd as far as I can see.

File: src/src/pages/capabilities/NewCapabilityDialog.js (hunk 6 above)
Comment (SEQUOIIA) [reply src/src/pages/capabilities/NewCapabilityDialog.js]: Addressed in https://github.com/dfds/selfservice-portal/pull/133/commits/cb4b6bfc44c8d7dff3c4ea8e1c66e2dc416c35bd

File: src/src/pages/capabilities/NewCapabilityDialog.js (hunk 5 above)
Comment (SEQUOIIA) [reply src/src/pages/capabilities/NewCapabilityDialog.js]: Adressed in Addressed in https://github.com/dfds/selfservice-portal/pull/133/commits/cb4b6bfc44c8d7dff3c4ea8e1c66e2dc416c35bd

File: src/src/pages/capabilities/NewCapabilityDialog.js (hunk 4 above)
Comment (SEQUOIIA) [reply src/src/pages/capabilities/NewCapabilityDialog.js]: Addressed in https://github.com/dfds/selfservice-portal/pull/133/commits/cb4b6bfc44c8d7dff3c4ea8e1c66e2dc416c35bd

File: src/src/pages/capabilities/NewCapabilityDialog.js (hunk 3 above)
Comment (SEQUOIIA) [reply src/src/pages/capabilities/NewCapabilityDialog.js]: Addressed in https://github.com/dfds/selfservice-portal/pull/133/commits/cb4b6bfc44c8d7dff3c4ea8e1c66e2dc416c35bd

File: src/src/hooks/Capabilities.js (hunk 2 above)
Comment (SEQUOIIA) [reply src/src/hooks/Capabilities.js]: Addressed in https://github.com/dfds/selfservice-portal/pull/133/commits/cb4b6bfc44c8d7dff3c4ea8e1c66e2dc416c35bd

File: src/src/components/DropDownMenu/index.js (hunk 1 above)
Comment (SEQUOIIA) [reply src/src/components/DropDownMenu/index.js]: Addressed in https://github.com/dfds/selfservice-portal/pull/133/commits/cb4b6bfc44c8d7dff3c4ea8e1c66e2dc416c35bd

**Pull request merged**